sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SymbolHelperTest.cs

```diff
@@ -73,193 +73,120 @@ public interface IInterface
 }
 ";
 
-        private ClassDeclarationSyntax baseClassDeclaration;
-        private ClassDeclarationSyntax derivedClassDeclaration1;
-        private ClassDeclarationSyntax derivedClassDeclaration2;
-        private InterfaceDeclarationSyntax interfaceDeclaration;
-        private SemanticModel semanticModel;
-        private SyntaxTree tree;
+        private SnippetCompiler testCode;
 
         [TestInitialize]
         public void Compile()
         {
-            var compilation = SolutionBuilder
-                .Create()
-                .AddProject(AnalyzerLanguage.CSharp, createExtraEmptyFile: false)
-                .AddSnippet(TestInput)
-                .GetCompilation();
-            this.tree = compilation.SyntaxTrees.First();
-            this.semanticModel = compilation.GetSemanticModel(this.tree);
-
-            this.baseClassDeclaration = this.tree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>()
-                .First(m => m.Identifier.ValueText == "Base");
-            this.derivedClassDeclaration1 = this.tree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>()
-                .First(m => m.Identifier.ValueText == "Derived1");
-            this.derivedClassDeclaration2 = this.tree.GetRoot().DescendantNodes().OfType<ClassDeclarationSyntax>()
-                .First(m => m.Identifier.ValueText == "Derived2");
-            this.interfaceDeclaration = this.tree.GetRoot().DescendantNodes().OfType<InterfaceDeclarationSyntax>()
-                .First(m => m.Identifier.ValueText == "IInterface");
+            this.testCode = new SnippetCompiler(TestInput, ignoreErrors: true, language: AnalyzerLanguage.CSharp);
```

Comment:
Refactored an existing test class to use _SnippetCompiler_. No other changes to the tests, or the product code they test.